r/tests/testthat/test-dplyr-funcs-datetime.R

```diff
@@ -2309,3 +2308,618 @@ test_that("build_formats() and build_format_from_order()", {
+          day(time1) == day(time2) &
+          hour(time1) == hour(time2) &
+          minute(time1) == minute(time2) &
+          second(time1) == second(time1))
+  }
+
+  base <- shifted_times$utc_rounded
+  expect_true(compare_local_times(shifted_times$syd_rounded, base))
+  expect_true(compare_local_times(shifted_times$adl_rounded, base))
+  expect_true(compare_local_times(shifted_times$mar_rounded, base))
+  expect_true(compare_local_times(shifted_times$kat_rounded, base))
+
+  base <- shifted_times$utc_floored
+  expect_true(compare_local_times(shifted_times$syd_floored, base))
+  expect_true(compare_local_times(shifted_times$adl_floored, base))
+  expect_true(compare_local_times(shifted_times$mar_floored, base))
+  expect_true(compare_local_times(shifted_times$kat_floored, base))
+
+  base <- shifted_times$utc_ceiling
+  expect_true(compare_local_times(shifted_times$syd_ceiling, base))
+  expect_true(compare_local_times(shifted_times$adl_ceiling, base))
+  expect_true(compare_local_times(shifted_times$mar_ceiling, base))
+  expect_true(compare_local_times(shifted_times$kat_ceiling, base))
+}
+
+test_that("timestamp rounding takes place in local time", {
+
+  tz_times %>% check_timezone_rounding_vs_lubridate(".001 second")
+  tz_times %>% check_timezone_rounding_vs_lubridate("second")
+  tz_times %>% check_timezone_rounding_vs_lubridate("minute")
+  tz_times %>% check_timezone_rounding_vs_lubridate("hour")
+  tz_times %>% check_timezone_rounding_vs_lubridate("day")
+  tz_times %>% check_timezone_rounding_vs_lubridate("week")
+  tz_times %>% check_timezone_rounding_vs_lubridate("month")
+  tz_times %>% check_timezone_rounding_vs_lubridate("quarter")
+  tz_times %>% check_timezone_rounding_vs_lubridate("year")
+
+  tz_times %>% check_timezone_rounding_for_consistency("second")
+  tz_times %>% check_timezone_rounding_for_consistency("minute")
+  tz_times %>% check_timezone_rounding_for_consistency("hour")
+  tz_times %>% check_timezone_rounding_for_consistency("day")
+  tz_times %>% check_timezone_rounding_for_consistency("week")
+  tz_times %>% check_timezone_rounding_for_consistency("month")
+  tz_times %>% check_timezone_rounding_for_consistency("quarter")
+  tz_times %>% check_timezone_rounding_for_consistency("year")
+
+  tz_times %>% check_timezone_rounding_for_consistency("7 seconds")
+  tz_times %>% check_timezone_rounding_for_consistency("7 minutes")
+  tz_times %>% check_timezone_rounding_for_consistency("7 hours")
+  tz_times %>% check_timezone_rounding_for_consistency("7 months")
+  tz_times %>% check_timezone_rounding_for_consistency("7 years")
+
+  tz_times %>% check_timezone_rounding_for_consistency("13 seconds")
+  tz_times %>% check_timezone_rounding_for_consistency("13 minutes")
+  tz_times %>% check_timezone_rounding_for_consistency("13 hours")
+  tz_times %>% check_timezone_rounding_for_consistency("13 months")
+  tz_times %>% check_timezone_rounding_for_consistency("13 years")
+
+})
```

Comment:
! we're almost at 3000 lines here. We may want to explore (doesn't have to be in this PR) breaking the file up--IMO files this big are hard to navigate (though I recognize that "too big" is subjective).

Reply:
I did some reorganisation in #13029, but the focus was not on reducing the lines per file, but on reducing cyclomatic complexity (there were at least a couple of higher - than the already increased to 26 - cyclomatic complexity instances). As a side-effect of that PR it actually increased the number of lines in test-dplyr-funcs-datetime.R as it moved all datetime tests into the same file.